webview/src/plots/hooks/useSetupSmoothPlot.ts

```diff
@@ -0,0 +1,50 @@
+import { useDispatch, useSelector } from 'react-redux'
+import { View } from 'react-vega'
+import { setSmoothPlotValue } from '../components/templatePlots/templatePlotsSlice'
+import { PlotsState } from '../store'
+
+interface VegaState {
+  signals?: { [name: string]: number | undefined }
+  data?: unknown
+}
+
+export const useSetupSmoothPlot = (id: string, isZoomedIn = false) => {
+  const dispatch = useDispatch()
+  const smoothPlotValues = useSelector(
+    (state: PlotsState) => state.template.smoothPlotValues
+  )
+
+  return (view: View) => {
+    const state = view.getState() as VegaState
+    const defaultValue = smoothPlotValues[id]
+
+    if (!state?.signals?.smooth) {
+      return
+    }
+
+    if (defaultValue) {
+      view.setState({
+        ...state,
+        signals: { ...state.signals, smooth: defaultValue }
+      })
+    }
+
+    if (isZoomedIn) {
+      return
+    }
+
+    const smoothRange = document.querySelector(
+      `[data-id="${id}"] input[name="smooth"]`
+    )
+    smoothRange?.addEventListener('change', (event: Event) => {
+      if (event.target) {
+        dispatch(
```

Comment:
[Q] Why not send the message here as well? That would mean we wouldn't have to `useEffect` in the other component which makes things harder to track.

Reply:
also don't have to send the entire object, could just send a partial update. I have a feeling this may fix the test as well.